gnd/src/main/java/com/google/android/gnd/GndApplication.kt

```diff
@@ -51,16 +51,17 @@ class GndApplication : MultiDexApplication(), Configuration.Provider {
 
             // Log failures when trying to do work in the UI thread.
             setStrictMode()
+        } else {
```

Comment:
Would the errors always be indicative of a broken state or are there potentially some scenarios that there would be an error that we could safely silently ignore?

Reply:
Yea I think that makes sense to crash immediately. When we're closer to launching it might make sense to catch errors and show a toast saying "Something went wrong, please try again" (depending on the severity of the error). That way users may be able to recover themselves from an error, and still be able to report the issue to us without the app crashing and potentially losing data.

Reply:
> Yea I think that makes sense to crash immediately during dev. When we're closer to launching it might make sense to catch errors and show a toast saying "Something went wrong, please try again" (depending on the severity of the error). That way users may be able to recover themselves from an error, and still be able to report the issue to us without the app crashing and potentially losing data.

If there are unhandled in the errors they could still be losing data, but since there won't be a force quit, we won't get Crashlytics reports about the issue with the actual stack trace of where the issue occurred. There were problems during the pilots where users reporting odd behaviors in the app, but we never got a log report about those issues. I'd be in favor of failing fast and fixing issues as they arise, and shifting to eating the errors iff that turns out to be unsustainable. Maybe we can try that for now and revert later as needed?

Reply:
Done